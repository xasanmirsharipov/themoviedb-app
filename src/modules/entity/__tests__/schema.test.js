import React from 'react';
import EntitySchema from '../schema';

describe('Entity Schema', () => {

	const entity = 'movie';

	it('should return schema', () => {

		const Schema = EntitySchema(entity, `${entity}Id`);

		expect(Schema).toEqual({
			"_getId": expect.any(Function),
			"_idAttribute": `${entity}Id`,
			"_key": entity,
			"_mergeStrategy": expect.any(Function),
			"_processStrategy": expect.any(Function),
			"schema": {}
		});

		expect(Schema).toMatchSnapshot();
	});

});