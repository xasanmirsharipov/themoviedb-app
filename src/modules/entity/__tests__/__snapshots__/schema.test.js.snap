// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Entity Schema should return schema 1`] = `
EntitySchema {
  "_getId": [Function],
  "_idAttribute": "movieId",
  "_key": "movie",
  "_mergeStrategy": [Function],
  "_processStrategy": [Function],
  "schema": Object {},
}
`;
